modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js

```diff
@@ -251,8 +203,78 @@ export default class Tile3DLayer extends CompositeLayer {
     );
   }
 
+  _create3DModelTileLayer(tileHeader) {
+    const {gltf, instances, cartographicOrigin, modelMatrix} = tileHeader.content;
+
+    const SubLayerClass = this.getSubLayerClass('scenegraph', ScenegraphLayer);
+
+    return new SubLayerClass(
+      {
+        _lighting: 'pbr'
+      },
+      this.getSubLayerProps({
+        id: 'scenegraph'
+      }),
+      {
+        id: `${this.id}-scenegraph-${tileHeader.id}`,
+        // Fix for ScenegraphLayer.modelMatrix, under flag in deck 7.3 to avoid breaking existing code
+        data: instances || [{}],
+        scenegraph: gltf,
+
+        coordinateSystem: COORDINATE_SYSTEM.METER_OFFSETS,
+        coordinateOrigin: cartographicOrigin,
+        modelMatrix,
+        getTransformMatrix: instance => instance.modelMatrix,
+        getPosition: instance => [0, 0, 0]
+      }
+    );
+  }
+
+  _createSimpleMeshLayer(tileHeader) {
+    const content = tileHeader.content;
+    const {attributes, modelMatrix, cartographicOrigin, texture} = content;
+    const {normals, texCoords} = attributes;
+    const positions = new Float32Array(attributes.positions.value.length);
+    for (let i = 0; i < positions.length; i += 3) {
```

Comment:
Why do you need to manually transform the positions? The layer’s modelMatrix should propagate in meter offsets mode.

Reply:
positions are `Float64`, can I construct a geometry with `Float64Array`?

Reply:
Casting it to Float32Array is fine. Why do you need to apply the modelMatrix?